sql/catalyst/src/main/scala/org/apache/spark/sql/types/YearMonthIntervalType.scala

```diff
@@ -54,6 +54,8 @@ class YearMonthIntervalType private() extends AtomicType {
   override def defaultSize: Int = 4
 
   private[spark] override def asNullable: YearMonthIntervalType = this
+
+  override def typeName: String = "year-month interval"
```

Comment:
according to the standard，year-month and day-time is italic，so I guess interval here is unnecessary.

Reply:
1. I haven't found any places in the SQL standard where `year-month` and `day-time` are used without the `interval(s)` word(s).
2. Actual type name is `INTERVAL`, `year-month` and `day-time` just define (sub-)classes of the type.
3. Since the `typeName()` method is used in error messages, I do believe we should leave `interval` in type names for readability.